recipe_box/src/main/java/com/spr/recipe_box/Service/Services/CatService.java

```diff
@@ -0,0 +1,35 @@
+package com.spr.recipe_box.Service.Services;
+
+import com.spr.recipe_box.Constant.Constants;
+import com.spr.recipe_box.Helper.StreamBytesHelper;
+import com.spr.recipe_box.Service.Interface.ICat;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import java.io.*;
+import java.net.URL;
+
+@Component
+public class CatService implements ICat {
+    StreamBytesHelper streamBytesHelper = new StreamBytesHelper();
+
+    public HttpEntity<byte[]> RandomCatPicture() throws IOException {
+        URL url = new URL(Constants.CAT_URL + Constants.CAT);
+        byte[] response = streamBytesHelper.getBytes(url);
+        HttpHeaders headers = getHttpHeaders(response);
+        return new HttpEntity<byte[]>(response, headers);
+    }
+
+    public HttpEntity<byte[]> RandomCatGif() throws IOException {
```

Comment:
RandomCatGif() looks like it shares a lot of the same code as RandomCatPicture(), you might want to create one method that can take in a string parameter so you could pass in Constants.Cat or Constants.Gif